### PR TITLE
(890) Add decorators to task pages

### PR DESCRIPTION
**Pull request by pezholio**
This builds on #382 to add decorators to our task page classes, this will allow to keep track of the fields used in each class, as well as their name and what task they are in. In a future PR, I'll then build on this to generate a schema from the classes.

This is a big PR, but hopefully is relatively simple to review, as it's mainly refactoring.

### CODE REVIEW

File: server/form-pages/apply/type-of-ap/pipeReferral.ts [hunk 1]
```diff
@@ -1,30 +1,50 @@
...
   }
 
-  constructor(body: Record<string, unknown>, private readonly application: Application) {
-    this.body = {
-      'opdPathwayDate-year': body['opdPathwayDate-year'] as string,
-      'opdPathwayDate-month': body['opdPathwayDate-month'] as string,
-      'opdPathwayDate-day': body['opdPathwayDate-day'] as string,
-      opdPathwayDate: body.opdPathwayDate as string,
-      opdPathway: body.opdPathway as YesOrNo,
+  constructor(public _body: Partial<PipeReferralBody>, private readonly application: Application) {}
```
Comment (bryangaledxw) [inline server/form-pages/apply/type-of-ap/pipeReferral.ts:26]: Can this be private? Ditto for other instance of "public _body"

File: server/form-pages/apply/access-and-healthcare/accessNeedsAdditionalAdjustments.test.ts [hunk 2]
```diff
@@ -96,9 +96,9 @@ describe('AccessNeedsAdditionalAdjustments', () => {
   })
 
   describe('body', () => {
-    it('should strip unknown attributes from the body', () => {
+    it('should apply the body', () => {
```
Comment (bryangaledxw) [inline server/form-pages/apply/access-and-healthcare/accessNeedsAdditionalAdjustments.test.ts:99]: "should set the body", just to match everywhere else?

File: server/form-pages/apply/access-and-healthcare/covid.test.ts [hunk 3]
```diff
@@ -21,7 +21,6 @@ describe('Covid', () => {
           fullyVaccinated: 'yes',
           highRisk: 'yes',
           additionalCovidInfo: 'some info',
-          somethingElse: 'ok',
```
Comment (bryangaledxw) [inline server/form-pages/apply/access-and-healthcare/covid.test.ts:24]: Change test name to "should set the body"?

File: server/form-pages/apply/further-considerations/arson.test.ts [hunk 4]
```diff
@@ -24,8 +25,8 @@ describe('Arson', () => {
   })
 
   describe('body', () => {
-    it('should strip unknown attributes from the body', () => {
-      const page = new Arson({ ...body, something: 'else' }, application)
+    it('should assign the body', () => {
```
Comment (bryangaledxw) [inline server/form-pages/apply/further-considerations/arson.test.ts:28]: "should set the body"  to match everywhere else?

File: server/form-pages/apply/check-your-answers/review.test.ts [hunk 5]
```diff
@@ -7,12 +7,12 @@ describe('Review', () => {
   const application = applicationFactory.build({})
 
   const body = {
-    reviewed: '1',
+    reviewed: 1,
   }
 
   describe('body', () => {
     it('should strip unknown attributes from the body', () => {
```
Comment (bryangaledxw) [inline server/form-pages/apply/check-your-answers/review.test.ts:14]: Change to "should set body"

File: server/form-pages/utils/decorators/page.decorator.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/ban-types */
```
Comment (bryangaledxw) [inline server/form-pages/utils/decorators/page.decorator.ts:1]: My preference would be for this to go by the specific lines we want these to apply to

**Review by bryangaledxw — APPROVED** [review]
Looks good, just minor comment

File: server/form-pages/utils/decorators/page.decorator.ts (hunk 6 above)
Comment (pezholio) [reply server/form-pages/utils/decorators/page.decorator.ts]: Yeah, good call 👍

**Pull request merged**